app/grandchallenge/evaluation/forms.py

```diff
@@ -985,3 +985,37 @@ def clean_ground_truth(self):
             raise ValidationError("Ground truth updating already in progress.")
 
         return ground_truth
+
+
+class AlgorithmInterfaceForPhaseCopyForm(Form):
+    phases = ModelMultipleChoiceField(
+        queryset=Phase.objects.none(),
+        label="Select the phases to copy the interfaces to",
+        widget=CheckboxSelectMultiple,
+    )
+
+    def __init__(self, *args, phase, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._phase = phase
+
+        self.fields["phases"].queryset = Phase.objects.filter(
+            challenge=phase.challenge,
+            submission_kind=phase.submission_kind,
+        ).exclude(pk=phase.pk)
+
+    def clean_phases(self):
+        phases = self.cleaned_data["phases"]
+
+        for phase in phases:
+            if phase.algorithm_interfaces_locked:
+                raise ValidationError(
+                    "You cannot copy algorithm interfaces to a phase "
+                    "where the algorithm interfaces are locked."
+                )
```

Comment:
It would probably help here to say which phase(s) is locked so that you can avoid selecting that phase when trying again.

Reply:
Done!